src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/show_request.cc

```diff
@@ -94,6 +99,125 @@ void ShowRequest::HandleMetadata(const char* field_id,
+                                                         << current_window_id_);
+      layout_change_required_ = true;
+      return true;
+    }
+    LOG4CXX_DEBUG(logger_, "Show Request: No Layout Change");
+    return false;
+  };
+
+  const auto set_day_color_scheme = [&app, this]() -> bool {
+    if (!template_config_.keyExists(strings::day_color_scheme)) {
+      return false;
+    }
+    if (app.day_color_scheme(current_window_id_).getType() !=
+            smart_objects::SmartType_Null &&
+        template_config_[strings::day_color_scheme] !=
+            app.day_color_scheme(current_window_id_)) {
+      // Color scheme param exists and has been previously set,
+      // hence do not allow color change
+      LOG4CXX_DEBUG(logger_, "Day Color Scheme change is rejected");
+      return false;
+    }
+    LOG4CXX_DEBUG(logger_, "Day Color Scheme change is allowed");
+    dcs_change_required_ = true;
+
+    return true;
+  };
+
+  const auto set_night_color_scheme = [&app, this]() -> bool {
+    if (!template_config_.keyExists(strings::night_color_scheme)) {
+      return false;
+    }
+    if (app.night_color_scheme(current_window_id_).getType() !=
+            smart_objects::SmartType_Null &&
+        template_config_[strings::night_color_scheme] !=
+            app.night_color_scheme(current_window_id_)) {
+      // Color scheme param exists and has been previously set,
+      // hence do not allow color change
+      LOG4CXX_DEBUG(logger_, "Night Color Scheme change is rejected");
+      return false;
+    }
+    LOG4CXX_DEBUG(logger_, "Night Color Scheme Change is allowed");
+    ncs_change_required_ = true;
+
+    return true;
+  };
+
+  const bool set_layout_result = set_window_layout();
+
+  if (set_layout_result) {
+    set_day_color_scheme();
+    set_night_color_scheme();
+    return true;
+  }
+
+  if (!template_config_.keyExists(strings::night_color_scheme) &&
+      !template_config_.keyExists(strings::day_color_scheme)) {
+    // In case current layout was not changed and day and night color
+    // schemes are absent in mobile message SDL has to forward message
+    // to HMI with the only layout even it was not changed
+    return true;
+  }
+
+  const bool set_schemes_result =
+      (set_day_color_scheme() && set_night_color_scheme());
+
+  return set_schemes_result;
+}
+
+void ShowRequest::ApplyTemplateConfigurationForApp(
+    mobile_apis::Result::eType result, application_manager::Application& app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (helpers::Compare<mobile_apis::Result::eType, helpers::EQ, helpers::ONE>(
+          result,
+          mobile_apis::Result::SUCCESS,
+          mobile_apis::Result::WARNINGS)) {
+    if (layout_change_required_) {
+      const std::string new_layout =
+          template_config_[strings::template_layout].asString();
+      LOG4CXX_DEBUG(logger_, "New layout : " << new_layout << " is applied");
+      app.set_window_layout(current_window_id_, new_layout);
```

Comment:
@mked-luxoft Looks that you can just add something like 
```cpp
if (template_config_.keyExists(strings::day_color_scheme)) {
  app.set_day_color_scheme(current_window_id_,
	                                        template_config_[strings::day_color_scheme]);
 }

if (template_config_.keyExists(strings::night_color_scheme)) {
  app.set_night_color_scheme(current_window_id_,
	                                           template_config_[strings::night_color_scheme]);
 }
 return;
```
to solve the problem described by @JackLivio

Reply:
applied in ebb5dcc5575fe835d3b64522cd8015a00af55cc9